polling_stations/apps/file_uploads/models.py

```diff
@@ -157,30 +157,55 @@ def pr_body(self):
         else:
             return f"**NB triggered from local machine**\n{message}"
 
-    def send_confirmation_email(self, user):
-        if user.email:
-            to = user.email
+    def send_confirmation_email(self):
+        server_env = getattr(settings, "SERVER_ENVIRONMENT", None)
+        # If we're in production, and the user has been deleted, return early.
+        # We don't want to send an email to a non-existent user and we already
+        # have github issues to track successful uploads
+        if server_env == "production" and self.upload_user == None:
+            return
+        # if we're in production, and the upload user exists, send them an email
+        elif server_env == "production" and self.upload_user != None:
+            to = self.upload_user.email
+            subject = f"Your file upload for {self.gss.short_name} ({self.election_date}) was successful"
+            message = render_to_string(
+                template_name="file_uploads/email/upload_confirmation.txt",
+                context={
+                    "subject": subject,
+                },
+            )
+            email = EmailMessage(
+                subject,
+                message,
+                settings.DEFAULT_FROM_EMAIL,
+                [to],
+                reply_to=[settings.DEFAULT_FROM_EMAIL],
+                headers={"Message-ID": subject},
+            )
+
+            return email.send()
+        # for all other environments, send the email to the default
+        # from email with a subject line that makes it clear
+        # we are not in production and testing is taking place
         else:
             to = settings.DEFAULT_FROM_EMAIL
-        subject = "Your file upload was successful"
-        message = render_to_string(
-            template_name="file_uploads/email/upload_confirmation.txt",
-            context={
-                "subject": subject,
-            },
-        )
-
-        email = EmailMessage(
-            subject,
-            message,
-            settings.DEFAULT_FROM_EMAIL,
-            [to],
-            [settings.DEFAULT_FROM_EMAIL],
-            reply_to=[settings.DEFAULT_FROM_EMAIL],
-            headers={"Message-ID": subject},
-        )
+            subject = f"**NB triggered from {server_env} instance** Your file upload for {self.gss.short_name} ({self.election_date}) was successful"
+            message = render_to_string(
+                template_name="file_uploads/email/upload_confirmation.txt",
+                context={
+                    "subject": subject,
+                },
+            )
+            email = EmailMessage(
+                subject,
+                message,
+                settings.DEFAULT_FROM_EMAIL,
+                [to],
+                reply_to=[settings.DEFAULT_FROM_EMAIL],
+                headers={"Message-ID": subject},
+            )
 
-        return email.send()
+            return email.send()
```

Comment:
This works, but we were chatting about how it could be DRY-ed up.
To achieve that I would re-write it as an `if`/`else` statement and just define `to` and `subject` which are the parts that change - rather than joining it to the previous `if`. We can then construct our `EmailMessage` instance.

I've also removed the `context` you're passing to `render_to_string` - as far as I can see `subject` is not being used in the template - this means it's quite readable just inlining the method call. 

Finally I don't think `return` is doing anything here - you don't use it in `polling_stations/apps/file_uploads/api.py`, so I've left it out. 

```python
# if we're in production, and the upload user exists, send them an email
if server_env == "production" and self.upload_user != None:
    to = self.upload_user.email
    subject = f"Your file upload for {self.gss.short_name} ({self.election_date}) was successful"
# for all other environments, send the email to the default
# from email with a subject line that makes it clear
# we are not in production and testing is taking place
else:
    to = settings.DEFAULT_FROM_EMAIL
    subject = f"**NB triggered from {server_env} instance** Your file upload for {self.gss.short_name} ({self.election_date}) was successful"

email = EmailMessage(
    subject,
    render_to_string(template_name="file_uploads/email/upload_confirmation.txt"),
    settings.DEFAULT_FROM_EMAIL,
    [to],
    reply_to=[settings.DEFAULT_FROM_EMAIL],
    headers={"Message-ID": subject},
)
email.send()
```

Reply:
Sorry, I thought we didn't want to send an email if the user didn't exist, even if in prod?

Reply:
This change suggests to me that if we are in prod and the user doesn't exist, an email will be sent to DC.

Reply:
I think you've deleted it, but before you had an `if` statement that caught that situation and returned early - meaning an email wouldn't be sent. I was suggesting replacing the block I selected to be replaced. so you'd have:
```python
if prod and not upload.user:
    return
if prod and upload.user:
    to = upload.user.email
    subject = "..."
else:
    to = DEFAULT
    subject = "TESTING..."
sendemail()
```

Reply:
> I would re-write it

here my 'it' was referring to the codeblock I selected. Not that clear, sorry.

Reply:
Ah ok, got it - sorry for the confusion. I've revised as agreed and updated tests.